packages/web3-rpc-providers/src/web3_provider_infura.ts

```diff
@@ -0,0 +1,169 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import {
+	EthExecutionAPI,
+	JsonRpcResponseWithResult,
+	Web3APIMethod,
+	Web3APIPayload,
+	Web3APIReturnType,
+	Web3APISpec,
+} from 'web3-types';
+import { ResponseError } from 'web3-errors';
+import { HttpProviderOptions } from 'web3-providers-http';
+import { Transport, Network, SocketOptions } from './types.js';
+import { Web3ExternalProvider } from './web3_provider.js';
+import { QuickNodeRateLimitError } from './errors.js';
+
+const isValid = (str: string) => str !== undefined && str.trim().length > 0;
+
+export class InfuraProvider<
+	API extends Web3APISpec = EthExecutionAPI,
+> extends Web3ExternalProvider {
+	// eslint-disable-next-line default-param-last
+	public constructor(
+		network: Network = Network.ETH_MAINNET,
+		transport: Transport = Transport.HTTPS,
+		token = '',
+		host = '',
+		providerConfigOptions?: HttpProviderOptions | SocketOptions,
+	) {
+		super(network, transport, token, host, providerConfigOptions);
+	}
+
+	public async request<
+		Method extends Web3APIMethod<API>,
+		ResultType = Web3APIReturnType<API, Method>,
+	>(
+		payload: Web3APIPayload<EthExecutionAPI, Method>,
+		requestOptions?: RequestInit,
+	): Promise<JsonRpcResponseWithResult<ResultType>> {
+		try {
+			return await super.request(payload, requestOptions);
+		} catch (error) {
+			if (error instanceof ResponseError && error.statusCode === 429) {
+				throw new QuickNodeRateLimitError(error);
+			}
+			throw error;
+		}
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	public getRPCURL(network: Network, transport: Transport, token: string, _host: string) {
+		let host = '';
+
+		switch (network) {
```

Comment:
what do you think about using map instead of switch, as https://github.com/web3/web3.js/pull/7260/files#diff-b3d860d130c02e49ecd127d925384fe54f14e0b93c94aec96cf2965781c9efa8R51

Reply:
done